Updates to ecosystem.md do clarify community roles (#34341)
facebook/react-native: ECOSYSTEM.md

* **Community Contributors:** Individuals who support projects in the [react-native-community](https://github.com/react-native-community) organization.

## Partners

Partners are companies that are significantly invested in React Native and demonstrate ownership. Informed by their use of React Native, they push for improvements of the core and/or the ecosystem around it. Examples of this may include large scale contributions to `react-native` or owning essential tools or libraries. 

Partners think of React Native as a product; they understand the trade offs that the project makes as well as future plans and goals. Together we shape the vision for React Native to make it the best way to build applications. 

Baseline responsibilities of partners include: 
* Attending monthly meeting
* Contributing to the release process. Examples include being a [community releaser](https://reactnative.dev/contributing/release-roles-responsibilities#release-role-2-community-releaser), testing new releases, technical support for release issues.
* Engage on the core contributor Discord

Our current partners and other areas of ownership:

> * **[Coinbase](TODO ):** TODO

Reviewer: Is Coinbase going to continue to invest in this? I know at least some of their internal RN champions were laid off earlier this year.

Author: Yea that's my question as well. We can discuss this in our next meeting. I think if I don't hear word back, will remove them from this list

Reviewer 2: 👋 Hi team! Brent here from Coinbase. We're down, but not out 😄 .

We're ramping back up, and hopefully y'all have been seeing Bruno in the weekly RN release meetings.

Reviewer 3: Also, it's worth mentioning we are getting close to open-source some of our tooling around measuring React Native performance.

For now, we shared a bit of a tease in this recent article https://blog.coinbase.com/performance-vitals-a-unified-scoring-system-to-guide-performance-health-and-prioritization-127f80268ee4, and more to come soon 😁

Reviewer 4: Additional work we are driving:
- Helping out migrating `@react-native-community/datetimepicker` to be Fabric compatible (and possibly others modules in the upcoming months)
- Investigating enabling the new architecture in our RN app, and will be happy to write a blogpost documenting the process and results once the project is completed
- Can help out with

Reviewer 4: Hey @lunaleaps, here our suggestion:

> **[Coinbase](https://coinbase.com/):** Builds world class crypto apps in React Native. Publishes [articles](https://blog.coinbase.com/tagged/react-native) to advertise React Native development. Supports OSS libraries on their migration to the new architecture.

> * **[Callstack](https://callstack.com/):** Maintains the [React Native CLI](https://github.com/react-native-community/react-native-cli) and organizes [React Native EU](https://react-native.eu/)

Reviewer 5: ```suggestion
* **[Callstack](https://callstack.com/):** Maintains the [React Native CLI](https://github.com/react-native-community/react-native-cli) and [other community libraries](https://github.com/callstack), organizes [React Native EU](https://react-native.eu/) and hosts [The React Native Show podcast](https://www.callstack.com/podcast-react-native-show)
```

* **[Expo](https://expo.dev/):** Builds [Expo Go and SDK](https://github.com/expo/expo), [Snack](https://snack.expo.dev/), and [Expo Application Services](https://expo.dev/eas). Maintains [React Native Directory](https://reactnative.directory/), stewards [React Navigation](https://reactnavigation.org/) along with other partners.

> * **[Infinite Red](https://infinite.red/):** Maintains the [ignite cli/boilerplate](https://github.com/infinitered/ignite), organizes [Chain React Conf](https://cr.infinite.red/)

Reviewer: IMO it's worth noting RN Radio podcast (5k+ subscribers) and RN Newsletter (13k+ subscribers), as the largest of their respective kinds in the RN community.

```suggestion
* **[Infinite Red](https://infinite.red/):** Maintains the [ignite cli/boilerplate](https://github.com/infinitered/ignite), organizes [Chain React Conf](https://cr.infinite.red/), hosts the [React Native Radio podcast](https://reactnativeradio.com), publishes the [React Native Newsletter](https://reactnativenewsletter.com)
```

* **[Meta](https://opensource.fb.com/):** Oversees the React Native product and maintains the [React Native core repo](https://reactnative.dev/) 
* **[Microsoft](https://twitter.com/ReactNativeMSFT):** Develops [React Native Windows](https://github.com/Microsoft/react-native-windows) and [React Native macOS](https://github.com/microsoft/react-native-macos) for building apps that target Windows and macOS; maintains [rnx-kit](https://github.com/microsoft/rnx-kit), [react-native-test-app](https://github.com/microsoft/react-native-test-app) and coordinates cross-companies efforts such as the [bundle working group](https://github.com/microsoft/rnx-kit/discussions/categories/bundle-working-group).
* **[Shopify](TODO):** TODO
* **[Software Mansion](https://swmansion.com/):** Maintain core infrastructure including JSC, Animated, and other popular third-party plugins and organizes [App.js Conf](https://appjs.co/)

> * **[Tesla](TODO ):** TODO

Reviewer 6: ```suggestion
* **[Tesla](https://www.tesla.com):** Maintains [React Native Camera Kit](http://github.com/teslamotors/react-native-camera-kit) and contributes to React Native core as well as various community libraries.
```

> 

Reviewer 7: ```suggestion
* **[Wix.com](wix.engineering/open-source):** Maintains a variety of React Native open source projects ([see all](https://github.com/orgs/wix/repositories?q=react-native)), including:  [Detox](https://wix.github.io/Detox/) end-to-end testing library for React Native apps, [RN UILib](https://wix.github.io/react-native-ui-lib/), [RN Navigation](https://wix.github.io/react-native-navigation/), [RN Calendars](https://wix.github.io/react-native-calendars/) and [RN Notifications](https://github.com/wix/react-native-notifications).

```

> * **[Wix.com](wix.engineering/open-source):** Maintains a variety of React Native open source projects ([see all](https://github.com/orgs/wix/repositories?q=react-native)), including:  [Detox](https://wix.github.io/Detox/) end-to-end testing library for React Native apps, [RN UILib](https://wix.github.io/react-native-ui-lib/), [RN Navigation](https://wix.github.io/react-native-navigation/), [RN Calendars](https://wix.github.io/react-native-calendars/) and [RN Notifications](https://github.com/wix/react-native-notifications).

Reviewer 7: ```suggestion
* **[Wix.com](https://wix.engineering/open-source):** Maintains a variety of React Native open source projects ([see all](https://github.com/orgs/wix/repositories?q=react-native)), including:  [Detox](https://wix.github.io/Detox/) end-to-end testing library for React Native apps, [RN UILib](https://wix.github.io/react-native-ui-lib/), [RN Navigation](https://wix.github.io/react-native-navigation/), [RN Calendars](https://wix.github.io/react-native-calendars/) and [RN Notifications](https://github.com/wix/react-native-notifications).
```

There are three types of stakeholders:

* **Partners:** Companies significantly invested in React Native and take responsibility for the React Native vision and community.
* **Core Contributors:** Individual people who contribute to the React Native project.
* **Community Contributors:** Individuals who support projects in the [react-native-community](https://github.com/react-native-community) organization.

## Partners

Partners are companies that are significantly invested in React Native and demonstrate ownership. Informed by their use of React Native, they push for improvements of the core and/or the ecosystem around it. Examples of this may include large scale contributions to `react-native` or owning an essential tools or libraries. 

Partners think of React Native as a product: they understand the trade offs that the project makes as well as future plans and goals. Together we shape the vision for React Native to make it the best way to build applications. 

> Baseline responsibilities of partners include: 

Reviewer 8: other potential options, just throwing there for brainstorming:
* help maintain the website and showcase (example of this right now: Simek from Expo)
* produce content/material/blogposts to showcase their usage of react-native on a consistent basis

Author: Great suggestions! One thing to clarify -- I was under the impression we really need more community commitments for release co-pilot. My concern is that if we make the baseline responsibilities too broad, then we might end up here again. The responsibilities of partners should be meeting baseline and in-addition demonstrating ownership. 

I think the points you listed fit well with demonstrating ownership-- we should also update Expo's list to include the website maintenance (cc @Simek if that's right by you)

Reviewer 9: @Simek has been contributing to the website off-the-clock, but we'll discuss whether we want to possibly make this an official responsibility. i won't have an answer soon and we shouldn't hold up this doc on it

Reviewer 8: yeah my point is basically related to what @brentvatne is saying in his other comment about ways of being involved:

> expo can certainly help out as a community releaser/release co-pilot in the future, although i wonder if it is useful to require that every partner contribute in this particular way. for example, our role in recent releases has been to provide our feedback from integrating the new release into our tooling and also give technical release support in the form of pull requests and reviews for pull requests. @Kudo has been handling this.

But yes, we'd like to get a couple more people that can be part of the release crew from the community perspective. I'm just concerned that some companies might shy away from becoming a partner if we set that as an "hard requirement"

Author: I see, I defer then to you guys what you think is necessary to state as baseline requirements -- or if there should be any at all. My impression is that it is difficult to get consistent commitment from the community for release support and leveraging our partners (which of course they're already doing) could help. 

We could make the baseline requirement still "support releases" but in a more broad definition as @brentvatne shared or get rid of it.

Reviewer 9: for this pass it may be appropriate to keep the definition broad, and we can refine that as we gain a better understanding of where different partners can contribute, and what we think should be a fair contribution level to balance the upsides of being a partner organization

* Attending monthly meeting

> * Contributing to the release process as a [release co-pilot](https://reactnative.dev/contributing/release-roles-responsibilities#release-role-2--release-copilot)

Reviewer 10: I don't know how we want to deal with cases when partners don't fulfil this requirement but we could decide on this when such a thing happens. I definitely think it's a great first step to make these requirements clear.

Author: Yea, I'm not sure either but agree consensus on these responsibilities is a start and if a partner isn't able to meet this, we can have a discussion about why and I think that'll also be helpful

Reviewer 9: the release co-pilot link anchor appears to be broken - i believe it should be updated to:

```suggestion
* Attending monthly meeting
* Contributing to the release process as a [community releaser](https://reactnative.dev/contributing/release-roles-responsibilities#release-role-2-community-releaser)
```

Reviewer 9: expo can certainly help out as a community releaser/release co-pilot in the future, although i wonder if it is useful to require that every partner contribute in this particular way. for example, our role in recent releases has been to provide our feedback from integrating the new release into our tooling and also give technical release support in the form of pull requests and reviews for pull requests. @kudo has been handling this.

* Engage on the core contributor Discord

Our current partners and other areas of ownership:
* **[Coinbase](TODO ):** TODO
* **[Callstack](https://callstack.com/):** Maintains the [React Native CLI](https://github.com/react-native-community/react-native-cli) and organizes [React Native EU](https://react-native.eu/)

> * **[Expo](https://expo.dev/):** Builds [expo](https://github.com/expo/expo) and [EAS](https://expo.dev/eas) on top of React Native to simplify app development

Reviewer 9: ```suggestion
* **[Expo](https://expo.dev/):** Builds [Expo Go and SDK](https://github.com/expo/expo), [Snack](https://snack.expo.dev/), and [Expo Application Services](https://expo.dev/eas). Maintains [React Native Directory](https://reactnative.directory/), stewards [React Navigation](https://reactnavigation.org/) along with other partners.
```

* **[Infinite Red](https://infinite.red/):** Maintains the [ignite cli/boilerplate](https://github.com/infinitered/ignite), organizes [Chain React Conf](https://cr.infinite.red/)
* **[Meta](https://opensource.fb.com/):** Oversees the React Native product and maintains the [React Native core repo](https://reactnative.dev/) 

> * **[Microsoft](http://aka.ms/reactnative):** Develops [React Native Windows](https://github.com/Microsoft/react-native-windows) and [React Native macOS](https://github.com/microsoft/react-native-macos) for building apps that target Windows and macOS

Reviewer 8: ```suggestion
* **[Microsoft](https://twitter.com/ReactNativeMSFT):** Develops [React Native Windows](https://github.com/Microsoft/react-native-windows) and [React Native macOS](https://github.com/microsoft/react-native-macos) for building apps that target Windows and macOS; maintains [rnx-kit](https://github.com/microsoft/rnx-kit), [react-native-test-app](https://github.com/microsoft/react-native-test-app) and coordinates cross-companies efforts such as the [bundle working group](https://github.com/microsoft/rnx-kit/discussions/categories/bundle-working-group).
```

Adding a few more things to our list of efforts

> * **[Shopify](TODO):** TODO

Reviewer 10: From our side, we could mention our opensource repositories like [flash-list](https://github.com/Shopify/flash-list) or [@shopify/react-native-skia](https://github.com/Shopify/react-native-skia) from William and Christian. We also sponsor SWM, among other things. @colinta, wdyt?

Author: ```suggestion
* **[Shopify](TODO):** Maintains [flash-list](https://github.com/Shopify/flash-list) and [@shopify/react-native-skia](https://github.com/Shopify/react-native-skia) and sponsorship of Software Mansion. 
```

Author: @fortmarek @colinta Does that suggestion look right? Is there a preferred link?

These are behaviors we have observed in our existing core contributors. They aren't strict rules but rather outline their usual responsibilities. We do not expect every core contributor to do all of the above things all the time. Most importantly, we want to create a supportive and friendly environment that fosters collaboration. Above all else, **we are always polite and friendly.**

Core contributor status is attained after consistently contributing and taking on the responsibilities outlined above and granted by other core contributors. Similarly, after a long period of inactivity (~6 months or more), a core contributor may be contacted to understand if they’re still interested in being part of the program.

> You can use this form <MISSING_LINK_TODO> to either:

Reviewer 8: really like the idea of having a form for people to become core contributors 👍

Author: @cortinico's idea!

* Apply yourself to become a Core Contributor. Make sure to include a list of valuable contributions you did to the React Native repository and ecosystem.
* Nominate someone to become a Core Contributor.

As a core contributor, you will have access to the core contributor Discord which is used for light-weight coordination and discussion. 

> **We aim to make contributing to React Native as easy and transparent as possible.** All important topics are handled through a [discussion or RFC process on GitHub](https://github.com/react-native-community/discussions-and-proposals). We are always looking for active, enthusiastic members of the React Native community to become core contributors.

Reviewer 8: not sure if here in the examples of where we aim for transparent and open communication you might want to add the releases WG (https://github.com/reactwg/react-native-releases) and the new arch WG (https://github.com/reactwg/react-native-new-architecture)

> **We aim to make contributing to React Native as easy and transparent as possible.** All important topics are handled through a [discussion or RFC process on GitHub](https://github.com/react-native-community/discussions-and-proposals). We are always looking for active, enthusiastic members of the React Native community to become core contributors.

Author: ```suggestion
**We aim to make contributing to React Native as easy and transparent as possible.** We have discussion groups dedicated to the [new architecture rollout](https://github.com/reactwg/react-native-new-architecture), [releases](https://github.com/reactwg/react-native-releases), and [general questions and proposals](https://github.com/react-native-community/discussions-and-proposals). We are always looking for active, enthusiastic members of the React Native community to become core contributors.
```
